python/lsst/ts/ess/common/sensor/base_sensor.py

```diff
@@ -0,0 +1,71 @@
+# This file is part of ts_ess_common.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["BaseSensor"]
+
+from abc import ABC, abstractmethod
+import logging
+from typing import List
+
+
+class BaseSensor(ABC):
+    """Base class for the different types of Sensors.
+
+    This class holds all common code for the sensors. Sensor specific code
+    (for instance temperature or wind sensor) needs to be implemented in a
+    sub-class.
+
+    Parameters
+    ----------
+    log: `logging.Logger`
+        The logger to create a child logger for.
+    num_channels: `int`, optional
+        The number of channels that the sensor will produce telemetry for. The
+        default value is 0 meaning that the number of channels is not variable.
+    """
+
+    @abstractmethod
+    def __init__(self, log: logging.Logger, num_channels: int = 0) -> None:
```

Comment:
Done. I did this because I am used to the way Java handles abstract classes and prevents initialization even if there are no abstract methods in them. I understand that Python handles this differently so I am adapting my coding style to that.